packages/jobs/lib/processor/handler.ts

```diff
@@ -22,13 +28,78 @@ async function abort(_task: OrchestratorTask): Promise<Result<void>> {
 }
 
 async function action(task: TaskAction): Promise<Result<JsonValue>> {
-    // TODO: Implement action processing
-    // Returning a successful result for now
-    return Ok({ taskId: task.id, dryrun: true });
+    const providerConfig = await configService.getProviderConfig(task.connection.provider_config_key, task.connection.environment_id);
+    if (providerConfig === null) {
+        return Err(`Provider config not found for connection: ${task.connection}`);
+    }
+
+    const syncRun = new syncRunService({
+        bigQueryClient,
+        integrationService,
+        recordsService,
+        slackService,
+        writeToDb: true,
+        logCtx: await logContextGetter.get({ id: String(task.activityLogId) }),
+        nangoConnection: task.connection,
+        syncName: task.actionName,
+        isAction: true,
+        syncType: SyncType.ACTION,
+        activityLogId: task.activityLogId,
+        input: task.input as object, // TODO: fix type after temporal is removed
+        provider: providerConfig.provider,
+        debug: false
+    });
+
+    const { error, response } = await syncRun.run();
+    if (error) {
+        return Err(error);
+    }
+    const res = jsonSchema.safeParse(response);
+    if (res.success) {
+        return Ok(res.data);
+    }
+    return Err(`Invalid action response format: ${response}. Action: ${JSON.stringify(task)}`);
 }
 
 async function webhook(task: TaskWebhook): Promise<Result<JsonValue>> {
-    // TODO: Implement action processing
-    // Returning an error for now
-    return Err(`Not implemented: ${JSON.stringify({ taskId: task.id })}`);
+    const providerConfig = await configService.getProviderConfig(task.connection.provider_config_key, task.connection.environment_id);
+    if (providerConfig === null) {
+        return Err(`Provider config not found for connection: ${task.connection}`);
+    }
+
+    const sync = await getSyncByIdAndName(task.connection.id, task.parentSyncName);
```

Comment:
ranting: descriptive names are great but this one is confusing since it's not using sync_id :D